Firmware/Board/v3/board.cpp

```diff
@@ -0,0 +1,416 @@
+        fet_thermistor_num_coeffs // num_coeffs
+    }, {
+#if HW_VERSION_MAJOR == 3 && HW_VERSION_MINOR >= 3
+        4, // adc_channel
+#else
+        1, // adc_channel
+#endif
+        &fet_thermistor_poly_coeffs[0], // coefficients
+        fet_thermistor_num_coeffs // num_coeffs
+    }
+};
+
+Motor motors[AXIS_COUNT] = {
+    {
+        &htim1, // timer
+        TIM_1_8_PERIOD_CLOCKS, // control_deadline
+        1.0f / SHUNT_RESISTANCE, // shunt_conductance [S]
+        m0_gate_driver, // gate_driver
+        m0_gate_driver // opamp
+    },
+    {
+        &htim8, // timer
+        (3 * TIM_1_8_PERIOD_CLOCKS) / 2, // control_deadline
+        1.0f / SHUNT_RESISTANCE, // shunt_conductance [S]
+        m1_gate_driver, // gate_driver
+        m1_gate_driver // opamp
+    }
+};
+
+Encoder encoders[AXIS_COUNT] = {
+    {
+        &htim3, // timer
+        {M0_ENC_Z_GPIO_Port, M0_ENC_Z_Pin}, // index_gpio
+        {M0_ENC_A_GPIO_Port, M0_ENC_A_Pin}, // hallA_gpio
+        {M0_ENC_B_GPIO_Port, M0_ENC_B_Pin}, // hallB_gpio
+        {M0_ENC_Z_GPIO_Port, M0_ENC_Z_Pin}, // hallC_gpio
+        &spi3_arbiter // spi_arbiter
+    },
+    {
+        &htim4, // timer
+        {M1_ENC_Z_GPIO_Port, M1_ENC_Z_Pin}, // index_gpio
+        {M1_ENC_A_GPIO_Port, M1_ENC_A_Pin}, // hallA_gpio
+        {M1_ENC_B_GPIO_Port, M1_ENC_B_Pin}, // hallB_gpio
+        {M1_ENC_Z_GPIO_Port, M1_ENC_Z_Pin}, // hallC_gpio
+        &spi3_arbiter // spi_arbiter
+    }
+};
+
+// TODO: this has no hardware dependency and should be allocated depending on config
+Endstop endstops[2 * AXIS_COUNT];
+
+SensorlessEstimator sensorless_estimators[AXIS_COUNT];
+Controller controllers[AXIS_COUNT];
+TrapezoidalTrajectory trap[AXIS_COUNT];
+OffboardThermistorCurrentLimiter motor_thermistors[AXIS_COUNT];
+
+std::array<Axis, AXIS_COUNT> axes{{
+    {
+        0, // axis_num
+        1, // step_gpio_pin
+        2, // dir_gpio_pin
+        (osPriority)(osPriorityHigh + (osPriority)1), // thread_priority
+        encoders[0], // encoder
+        sensorless_estimators[0], // sensorless_estimator
+        controllers[0], // controller
+        fet_thermistors[0], // fet_thermistor
+        motor_thermistors[0], // motor_thermistor
+        motors[0], // motor
+        trap[0], // trap
+        endstops[0], endstops[1], // min_endstop, max_endstop
+    },
+    {
+        1, // axis_num
+#if HW_VERSION_MAJOR == 3 && HW_VERSION_MINOR >= 5
+        7, // step_gpio_pin
+        8, // dir_gpio_pin
+#else
+        3, // step_gpio_pin
+        4, // dir_gpio_pin
+#endif
+        osPriorityHigh, // thread_priority
+        encoders[1], // encoder
+        sensorless_estimators[1], // sensorless_estimator
+        controllers[1], // controller
+        fet_thermistors[1], // fet_thermistor
+        motor_thermistors[1], // motor_thermistor
+        motors[1], // motor
+        trap[1], // trap
+        endstops[2], endstops[3], // min_endstop, max_endstop
+    },
+}};
```

Comment:
is it possible to do the struct initialization by name thing like `.stuff = bla` ?

Reply:
Unfortunately no because these are initializer lists that get applied as arguments to the `Axis` constructor.